src/statemachine.rs

```diff
@@ -613,27 +672,61 @@ impl StateMachine {
                 };
 
                 info!("Device {:?} continues with the signing process", dev.id());
-                // TODO(MS): This is wasteful, but the current setup with read only-functions doesn't allow me
-                //           to modify "params" directly.
-                let mut getassertion = params.clone();
-                // First check if extensions have been requested that are not supported by the device
-                if params.extensions.hmac_secret.is_some() {
-                    if let Some(auth) = dev.get_authenticator_info() {
-                        if !auth.supports_hmac_secret() {
-                            callback.call(Err(AuthenticatorError::UnsupportedOption(
-                                UnsupportedOption::HmacSecret,
-                            )));
-                            return;
-                        }
+
+                // We need a copy of the arguments for this device
+                let args = args.clone();
+
+                if let Some(info) = dev.get_authenticator_info() {
+                    // Check if extensions have been requested that are not supported by the device
+                    if args.extensions.hmac_secret.is_some() && !info.supports_hmac_secret() {
+                        callback.call(Err(AuthenticatorError::UnsupportedOption(
+                            UnsupportedOption::HmacSecret,
+                        )));
+                        return;
+                    }
+                } else {
+                    // Check that the request can be processed by a CTAP1 device.
+                    // See CTAP 2.1 Section 10.3. Some additional checks are performed in
+                    // GetAssertion::RequestCtap1
+                    if args.user_verification_req == UserVerificationRequirement::Required {
+                        callback.call(Err(AuthenticatorError::UnsupportedOption(
+                            UnsupportedOption::UserVerification,
+                        )));
+                        return;
+                    }
+                    if args.allow_list.is_empty() {
+                        callback.call(Err(AuthenticatorError::UnsupportedOption(
+                            UnsupportedOption::AllowList,
```

Comment:
Maybe call the error `UnsupportedOption::EmptyAllowList`? Otherwise it sounds like having an allow list is unsupported, which may be confusing, when this error is printed somewhere.

Reply:
Sounds good.